### PR TITLE
Make code more compatible for Pytorch version 0.4.0 and above

### PR DESCRIPTION
**Pull request by varshaneya**
The following changes were made from the base branch:

- From Pytorch version 0.4 onwards, Variable class is deprecated. So removed all calls to Variable.

- Increased readability in line 46 by replacing `len(seq[:,0])` with `seq.shape[0]`. Both means the same.

- For lines 186 -- 191, transposing a 3D tensor throws an error. So squeeze operation was done before transpose rather than doing after.

- For lines 219 -- 226, the use of `detach()` is unnecessary and superfluous as these tensors are not part of the main computation graph and the variable `batch` has `requires_grad` to be `False`.

### CODE REVIEW

**Comment by alexis-jacq** [comment]
Oh thanks for this, I definitely had to remove Variables from the codes for the last version, but I couldn't find the time.
It seems ok, I merge your modifs.
